### ISSUE TITLE
create install-config fails when project_name is not configured

### ISSUE THREAD

**Issue by chrigl** [openshift/installer#2710]
# Version

```console
$ openshift-install version
./openshift-install v4.2.5
built from commit 425e4ff0037487e32571258640b39f56d5ee5572
release image quay.io/openshift-release-dev/ocp-release@sha256:bac62983757570b9b8f8bc84c740782984a255c16372b3e30cfc8b52c0a187b9
```

# Platform:

openstack

# What happened?

`./openshift-install --dir=initial create install-config foo` crashed with  Authentication error after already talking successfully to the OpenStack API

```
> ./openshift-install --dir=foo create install-config
? SSH Public Key /Users/chris/.ssh/id_rsa.pub
? Platform openstack
? Cloud imke-test
? ExternalNetwork provider
? APIFloatingIPAddress 185.116.247.101
? FlavorName m1.medium
FATAL failed to fetch Install Config: failed to fetch dependency of "Install Config": failed to fetch dependency of "Base Domain": failed to generate asset "Platform": Authentication failed
```

# What you expected to happen?

Initial config will be written.

# How to reproduce it (as minimally and precisely as possible)?

Create a valid and working `clouds.yaml`, but do configure `project_id` rather than `project_name`.
Run `./openshift-install --dir=foo create install-config`

# Anything else we need to know?

The code in `GetServiceCatalog` only accepts `ProjectName`
https://github.com/openshift/installer/blob/master/pkg/types/openstack/validation/realvalidvaluesfetcher.go#L150

The code could be simplified to the following
```
func (f realValidValuesFetcher) GetServiceCatalog(cloud string) ([]string, error) {
        opts := &clientconfig.ClientOpts{
                Cloud: cloud,
        }

        conn, err := clientconfig.NewServiceClient("identity", opts)
        if err != nil {
                return nil, err
        }

        authResult := conn.GetAuthResult()
        auth, ok := authResult.(tokens.CreateResult)
        if !ok {
                return nil, errors.New("unable to extract service catalog")
        }

        allServices, err := auth.ExtractServiceCatalog()
        if err != nil {
                return nil, err
        }

        serviceCatalogNames := make([]string, len(allServices.Entries))
        for i, service := range allServices.Entries {
                serviceCatalogNames[i] = service.Name
        }

        return serviceCatalogNames, nil
}
```

Because the original code already expects keystone v3, the example code also only works with v3.
`conn.GetAuthResult()` already includes the ServiceCatalog. So there is no need to construct an own `tokens.Scope`, but rather let `gophercloud` do the heavy lifting of finding out the correct combination of parameters in `clouds.yaml` ;)

# References

**Comment by chrigl** [comment]
For anyone running in this error and reading this: You just need to configure `project_name` in your `clouds.yaml`.

**Comment by pierreprinetti** [comment]
Added as [Bug 1776272](https://bugzilla.redhat.com/show_bug.cgi?id=1776272), thanks for reporting.

**Comment by pierreprinetti** [comment]
The [fix](https://github.com/openshift/installer/pull/2726) is merged to master and will be shipped with 4.4. Do you have the chance to confirm it's working for you?

**Comment by mandre** [comment]
The fix was merged and will be available in the 4.4 release.
/close
